### PR TITLE
Error logging for run() called prior to train()

### PR DESCRIPTION
**Pull request by perkyguy**
1621294-low_hanging_fruit_super

## Description

I added a function `isRunnable` as a `get` function, which will check for the side-effects of `train` being run. This 
In `run`, I added a call to 

## Motivation and Context

[issue](https://github.com/BrainJS/brain.js/issues/135)

## How Has This Been Tested?

Used the examples, and commented out tried doing a call to `run` prior to the `train` calls. Then `train`, and `run` again. The first time through, there should be an error message (and a `null` result), the second time through, it should work as expected. 

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Author's Checklist:

- [x] My code focuses on the main motivation and avoids scope creep.
- [x] My code passes current tests and adds new tests where possible.
- [x] My code is [SOLID](https://en.wikipedia.org/wiki/SOLID_(object-oriented_design)) and [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself).
- [x] I have updated the documentation as needed.

## Reviewer's Checklist:
- [ ] I kept my comments to the author positive, specific, and productive.
- [ ] I tested the code and didn't find any new problems.
- [ ] I think the motivation is good for the project.
- [ ] I think the code works to satisfies the motivation.

### CODE REVIEW

File: src/neural-network.js [hunk 1]
```diff
@@ -119,12 +119,42 @@ export default class NeuralNetwork {
...
+    const checkFns = [
+      'sizes',
+      'outputLayer',
+      'biases',
+      'weights',
+      'outputs',
+      'deltas',
+      'changes',
+      'errors',
+    ].map(c => this[c]).every(fn => fn !== null);
```
Comment (robertleeplummerjr) [inline src/neural-network.js]: If you change this to a filter, and check for length, you'd have the properties that failed

File: src/neural-network.js [hunk 2]
```diff
@@ -119,12 +119,42 @@ export default class NeuralNetwork {
...
+
+    if(!checkFns){
+      console.error('Some settings have not been initialized correctly, did you run train()?');
```
Comment (robertleeplummerjr) [inline src/neural-network.js]: which you could then use here, for a more specific error

File: src/recurrent/rnn.js [hunk 3]
```diff
@@ -293,6 +293,21 @@ export default class RNN {
     this.ratioClipped = numClipped / numTot;
   }
 
+
+  /**
+   *
+   * @returns boolean
+   */
+  get isRunnable(){
+    if(this.model.equations.length === 0){
```
Comment (robertleeplummerjr) [inline src/recurrent/rnn.js:302]: nice

File: src/neural-network.js (hunk 2 above)
Comment (perkyguy) [reply src/neural-network.js]: I like that much better. Check out the new implementation

File: src/recurrent/rnn.js (hunk 3 above)
Comment (perkyguy) [reply src/recurrent/rnn.js:302]: This is one of the defining side effects of `train`, correct?

File: src/neural-network.js (hunk 1 above)
Comment (perkyguy) [reply src/neural-network.js]: added, that is much cleaner/better

**Review by robertleeplummerjr — APPROVED** [review]
love it!

**Pull request merged**